src/writer/blocks/shared_blueprint_registry.py

```
@@ -0,0 +1,388 @@
+    blueprint_refs: Set[str] = set()
+    shared_blueprint_refs: Set[str] = set()
+    
+    # Pattern for state variable references: @{variable_name}
+    state_var_pattern = re.compile(r'@\{([^}]+)\}')
+    
+    for comp in components:
+        comp_type = comp.get("type", "")
+        content = comp.get("content", {})
+        
+        # Scan all string values in content for state variable references
+        for key, value in content.items():
```

Ruff (0.14.5)

Loop control variable `key` not used within loop body

Rename unused `key` to `_key`

(B007)